### PR TITLE
cloudflared: fix for new API responses, json nested under .results and .errors

### PR DESCRIPTION
**Pull request by pekovac**
Latest deployments are failing on `null` values in `CF_TUNNEL_ID` , `CREDENTIALS_FILE` and configuration files being rendered useless, containing only `null` value.

Logs:
```
 [cont-init.d] 98-cloudflared-config: executing... 
 **** Cloudflared setup script init... ****
 **** Checking cloudflared setup script requirements... ****
 **** Linux architecture found: amd64 ****
 **** Linux distro found: alpine ****
 **** Checking for cloudflared setup script dependencies... ****
 **** Temporarily installing /tmp/yq... ****
 **** jq already installed, skipping... ****
 **** curl already installed, skipping... ****
 **** Installing cloudflared...****
 **** Moving /cloudflared/cloudflared-amd64 to /usr/local/bin/cloudflared... ****
 **** Deleting tmp /cloudflared dir... ****
 **** Cloudflared installed ****
 cloudflared version 2021.10.5 (built 2021-10-25-2039 UTC)
 **** Checking for optional cloudflare tunnel parameters... ****
 **** Cloudflare tunnel parameters found, starting cloudflare tunnel setup... ****
 **** Creating cloudflare tunnel(nginx) via API... ****
 {
   "success": true,
   "messages": [],
   "errors": [],
   "result": {
     "id": "xxxxxxx-zzzzzz-yyyy-a8a7-xxxxxxxxxxx",
     "account_tag": "xxxxxxxxxxxxxxxxxxxxxxxxxxx",
     "created_at": "2021-11-01T04:21:51.464282Z",
     "deleted_at": null,
     "name": "nginx",
     "connections": [],
     "conns_active_at": null,
     "conns_inactive_at": "2021-11-01T04:21:51.464282Z",
     "tun_type": "cfd_tunnel",
     "metadata": {},
     "status": "inactive",
     "credentials_file": {
       "AccountTag": "xxxxxxxxxxxxxxxxxxxxxxxxxxx",
       "TunnelID": "xxxxxxx-zzzzzz-yyyy-a8a7-xxxxxxxxxxx",
       "TunnelName": "nginx",
       "TunnelSecret": "WjRSaEU2YU40ODdCWFVEZFp5QThtcFFuaGh4QzlNWUsjb3B0aW9uYWwK"
     }
   }
 }
 **** Saving cloudflare tunnel(nginx) credentials json... ****
 null
 **** Cloudflare tunnel(nginx) credentials saved to /etc/cloudflared/null.json ****
 **** Generating config.yml for cloudflare tunnel(nginx)... ****
 tunnel: null
 credentials-file: /etc/cloudflared/null.json
 no-autoupdate: true
 ingress:
   - hostname: nginx.example.com
     service: hello_world
   - service: http_status:404
 **** Config for cloudflare tunnel(nginx) saved to /etc/cloudflared/config.yml ****
 **** Validating cloudflared tunnel rules... ****
 Validating rules from /etc/cloudflared/config.yml
 OK
 **** Updating cloudflare zone... ****
 **** Searching zone for hostname(nginx.example.com) via API... ****
 **** Updating existing CNAME for hostname(nginx.example.com) via API... ****
 {
   "result": {
     "id": "d133c8eafff1e2bddff702311e39d0f6",
     "zone_id": "00e22252e091dcaa6b10d98251ff76a6",
     "zone_name": "example.com",
     "name": "nginx.example.com",
     "type": "CNAME",
     "content": "null.cfargotunnel.com",
     "proxiable": true,
     "proxied": true,
     "ttl": 1,
     "locked": false,
     "meta": {
       "auto_added": false,
       "managed_by_apps": false,
       "managed_by_argo_tunnel": false
     },
     "created_on": "2021-11-01T02:25:48.507188Z",
     "modified_on": "2021-11-01T04:16:41.696773Z"
   },
   "success": true,
   "errors": [],
   "messages": []
 }
 **** Cleaning up cloudflared setup script dependencies if required... ****
 **** Uninstalling /tmp/yq... ****
 **** Cloudflared setup script done, exiting... ****
 [cont-init.d] 98-cloudflared-config: exited 0.
 [cont-init.d] 99-custom-files: executing... 
 [custom-init] no custom files found exiting...
 [cont-init.d] 99-custom-files: exited 0.
 [cont-init.d] done.
 [services.d] starting services
 [services.d] done.
 "cloudflared tunnel run" requires the ID or name of the tunnel to run as the last command line argument or in the configuration file.
```

Content of the files:

```
docker exec -ti nginx bash
root@548657ff7c7c:/# cd /etc/cloudflared/
root@548657ff7c7c:/etc/cloudflared# ls -la 
total 16
drwxr-xr-x 1 abc  users   38 Nov  1 04:16 .
drwxr-xr-x 1 root root  4096 Nov  1 04:16 ..
-rw-r--r-- 1 root root   174 Nov  1 04:16 config.yml
-rw-r--r-- 1 root root     4 Nov  1 04:16 null.json
root@548657ff7c7c:/etc/cloudflared# cat config.yml 
tunnel: null
credentials-file: /etc/cloudflared/null.json
no-autoupdate: true

ingress:
  - hostname: nginx.example.com
    service: hello_world
  - service: http_status:404
root@548657ff7c7c:/etc/cloudflared# cat null.json 
nullroot@548657ff7c7c:/etc/cloudflared# 
```

### CODE REVIEW

**Comment by aptalca** [comment]
@spunkie can you review?

**Comment by Spunkie** [comment]
I'm mostly off-grid traveling this past few weeks but I'll test tonight/tomorrow when I move where I can actually pickup some stable internet.

File: root/etc/cont-init.d/98-cloudflared-config [hunk 1]
```diff
@@ -99,24 +99,27 @@ if [[ ${#CF_ACCOUNT_ID} -gt 0 ]] && [[ ${#CF_API_TOKEN} -gt 0 ]] && [[ ${#CF_TUN
...
       echo "**** Fetching existing cloudflare tunnel(${CF_TUNNEL_NAME}) via API... ****"
-      CF_TUNNEL_ID=$(echo ${JSON_RESULT} | jq -rc ".[].id")
+      CF_TUNNEL_ID=$(echo ${JSON_RESULT} | jq -rc ".result[].id")
       JSON_RESULT=$(curl -sX \
         GET "https://api.cloudflare.com/client/v4/accounts/${CF_ACCOUNT_ID}/tunnels/${CF_TUNNEL_ID}?" \
           -H "Authorization: Bearer ${CF_API_TOKEN}" \
           -H "Content-Type: application/json")
 
-      JSON_RESULT=$(echo ${JSON_RESULT} | jq -rc ". + {\"credentials_file\": {\"AccountTag\": \"${CF_ACCOUNT_ID}\",\"TunnelID\": \"${CF_TUNNEL_ID}\",\"TunnelName\": \"${CF_TUNNEL_NAME}\",\"TunnelSecret\": \"${CF_TUNNEL_SECRET}\"}}")
+      JSON_RESULT=$(echo ${JSON_RESULT} | jq -rc ". |= .+ {\"credentials_file\": {\"AccountTag\": \"${CF_ACCOUNT_ID}\",\"TunnelID\": \"${CF_TUNNEL_ID}\",\"TunnelName\": \"${CF_TUNNEL_NAME}\",\"TunnelSecret\": \"${CF_TUNNEL_SECRET}\"}}")
```
[the remark this answers is not in the document]
Comment (pekovac) [reply root/etc/cont-init.d/98-cloudflared-config:108]: @Spunkie jq pipes the output and tries to add the `credentials_file` to the top of the result. With the POST curl when creating new tunnel, this original jq added `credentials_file` inside of `result` (as seen in the log) , while when getting with GET when tunnel already exists, `credentials_file` was added to the top of the structure. 

Still it does not behave as expected, behaves as original statement and it  does not add `credentials_file` to the top when getting `JSON_RESPONSE`, so the value of `CREDENTIALS_FILE=null` 

This is then handled in line 113:
```bash
    CREDENTIALS_FILE=$(echo ${JSON_RESULT} | jq -rc ".credentials_file")
    if [ $CREDENTIALS_FILE = "null" ]; then # when created through POST, credentials_file is part of .results (maybe incomplete json response), check where it is
      CREDENTIALS_FILE=$(echo ${JSON_RESULT} | jq -rc ".result.credentials_file")
    fi
```

In short, this change behaves as original in this case, just pipes the output and can be reverted.

**Review by Spunkie — APPROVED** [review]
Can confirm my old previously working test code is showing the same reported issue. This fix for it seems good to me. :+1:

**Comment by aptalca** [comment]
@pekovac thanks so much for the PR

**Comment by Spunkie** [comment]
Did some e2e testing with the fresh docker-mod build and it works. Thanks for the fix @pekovac :tada:
